File: frontend/src/services/clients/pessoaService.js
import apiService from "./apiService";

export const listarPessoas = async (page = 0, size = 5) => {
  try {
    const response = await apiService.get("/senai-backend/v1/pessoa/lista", { params: { page, size } });
    return response.data;
  } catch (error) {
    throw error.response?.data || error.message;
  }
};

export const criarPessoa = async (pessoaData) => {
  try {
    const response = await apiService.post("/senai-backend/v1/pessoa", pessoaData);
    return response.data;
  } catch (error) {
    throw error.response?.data || error.message;
  }
};

export const atualizarPessoa = async (idPessoa, pessoaData) => {
  try {
    const response = await apiService.put(`/senai-backend/v1/pessoa/${idPessoa}`, pessoaData);
    return response.data;
  } catch (error) {
    throw error.response?.data || error.message;
  }
};

export const deletarPessoa = async (idPessoa) => {
  try {
    const response = await apiService.delete(`/senai-backend/v1/pessoa/${idPessoa}`);
    return response.data;
  } catch (error) {
    throw error.response?.data || error.message;
  }
};

File: backend/src/main/java/com/senai/backend/service/PessoaService.java
package com.senai.backend.service;

import com.senai.backend.api.request.AtualizarPessoaRequest;
import com.senai.backend.api.request.CriarPessoaRequest;
import com.senai.backend.api.response.PessoaResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

@Service
public interface PessoaService {

    PessoaResponse criarPessoa(CriarPessoaRequest request);
    Page<PessoaResponse> listarPessoas(Pageable pageable);
    PessoaResponse atualizarPessoa(Integer idPessoa, AtualizarPessoaRequest request);
    void deletarPessoa(Integer idPessoa);
}
